cmd/ec2.go

```diff
@@ -15,21 +19,52 @@ var ec2Cmd = &cobra.Command{
 	Short: "Print ec2 related information",
 	Long: `For example,
 		awsctl get ec2`,
-	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("ec2 called")
-	},
+	Run: getEC2command,
 }
 
-func init() {
-	getCmd.AddCommand(ec2Cmd)
+type EC2DesAPI interface {
+	DescribeInstances(ctx context.Context,
+		params *ec2.DescribeInstancesInput,
+		optFns ...func(*ec2.Options)) (*ec2.DescribeInstancesOutput, error)
+}
+
+// Returns info about ec2 instance
+func GetEc2Info(ctx context.Context, input *ec2.DescribeInstancesInput, api EC2DesAPI) (*ec2.DescribeInstancesOutput, error) {
+
+	return api.DescribeInstances(ctx, input)
+
+}
+
+func getEC2command(cmd *cobra.Command, args []string) {
+
+	ctx := context.TODO() // Unclear Context
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		log.Fatal("Config Error Occured", err)
+	}
 
-	// Here you will define your flags and configuration settings.
+	client := ec2.NewFromConfig(cfg)
 
-	// Cobra supports Persistent Flags which will work for this command
-	// and all subcommands, e.g.:
-	// ec2Cmd.PersistentFlags().String("foo", "", "A help for foo")
+	input := &ec2.DescribeInstancesInput{} // Res is DryRunOperation if everything is OK
 
-	// Cobra supports local flags which will only run when this command
-	// is called directly, e.g.:
-	// ec2Cmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	info, err := GetEc2Info(ctx, input, client)
+	if err != nil {
+		log.Fatal("Error Occured while retrieving Information", err)
+	}
+
+	for _, r := range info.Reservations {
```

Comment:
Since we are taking inspiration from kubectl, we need to print information in table format.

you can refer to `awsctl get s3` on how to use tabwriter library for tabular output.

Reply:
Okay, I will check. I didn't knew about this tabwriter library I added "\t", thanks @surajnarwade for sharing